### DISCUSSION TITLE
Schema and query codegen

### DISCUSSION THREAD

**Discussion by helderco** [strawberry-graphql/strawberry#2246] [General] [answered]
I'm interested in knowing the progress on this. Not just where you're at but what you've come up with so far and what ideas you have to improve it.

In [Query Codegen](https://strawberry.rocks/docs/codegen/query-codegen#query-codegen):

> Schema codegen will be supported in future releases.

Has anyone started on schema codegen? Any thoughts on design? Is it going to be plain objects or strawberry types?

> We are testing the query codegen in order to come up with a nice API.

Does this mean testing the current released implementation to get feedback from users or is there progress somewhere (ideas, code...)?

Thanks! 🙏

**Comment by jhnnsrs** [comment, marked as answer]
Have developed a python project "turms" that can do both query and schema codegen;)

https://github.com/jhnnsrs/turms/pull/44

Adds support for strawberry. You can even use this for schema first approaches, where you migrate changes into the
code generated code that you have altered to fit your need (implemented resolvers and additional fields will not change
where as documentation and new types would be added).

Hope that helps ;) will soon push this to main!
